packages/types/src/type/ExtrinsicEra.ts

```diff
@@ -2,33 +2,189 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
+import Enum from '../codec/Enum';
+import { assert, hexToU8a, isHex, isU8a, isObject, u8aToBn } from '@polkadot/util';
+import Tuple from '../codec/Tuple';
+import U8 from '../primitive/U8';
 import { AnyU8a } from '../types';
-
-import { u8aToU8a } from '@polkadot/util';
-
 import U8a from '../codec/U8a';
 
 /**
  * @name ExtrinsicEra
  * @description
  * The era for an extrinsic, indicating either a mortal or immortal extrinsic
  */
-export default class ExtrinsicEra extends U8a {
+export default class ExtrinsicEra extends Enum {
+  constructor (value?: any) {
+    super({ ImmortalEra, MortalEra }, ExtrinsicEra.decodeExtrinsicEra(value));
+  }
+
+  private static decodeExtrinsicEra (value: MortalMethod | Uint8Array | string): Uint8Array | Object | undefined {
+    if (!value) {
```

Comment:
This can be replaced by making the signature `value: MortalMethod | Uint8Array | string = new Uint8Array([0])` (Or even `= new Uint8Array()` and just adjusting the check lower down to be 

```js
else if (isU8a(value)) {
      if (!value.length || value[0] === 0) {
```